velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -91,13 +91,23 @@ class StringFunctionsTest : public FunctionBaseTest {
       int stringVectorIndex = 0) {
     auto row = makeRowVector(args);
     auto stringVector = args[stringVectorIndex];
-    int refCountBeforeEval =
-        bufferRefCounts(stringVector->asFlatVector<StringView>());
+    auto flatStringArg = stringVector->asFlatVector<StringView>();
+
+    int refCountBeforeEval = bufferRefCounts(flatStringArg);
     auto result = evaluate<FlatVector<StringView>>(query, row);
 
-    int refCountAfterEval =
-        bufferRefCounts(stringVector->asFlatVector<StringView>());
-    EXPECT_EQ(refCountAfterEval, 2 * refCountBeforeEval) << "at " << query;
+    int refCountAfterEval = bufferRefCounts(flatStringArg);
```

Comment:
@yabinma : Why did you change the EXPECT condition ? Is EXPECT_EQ(refCountAfterEval, 2 * refCountBeforeEval) not valid anymore ?

Reply:
@aditi-pandit ,  yes,  the previous EXPECT_EQ does not work anymore after adding test case for substring(evaluateSubstr is called twice after the new substring case added, buffer reference count is difference with twice calls).
https://github.com/facebookincubator/velox/pull/13220#discussion_r2077360039